opentelemetry/index.ts

```diff
@@ -0,0 +1,87 @@
+import { DiagConsoleLogger, DiagLogLevel, diag } from '@opentelemetry/api'
+import { NodeSDK } from '@opentelemetry/sdk-node'
+import axios from 'axios'
+import { config } from 'dotenv'
+import path from 'path'
+import { DefaultInstrumentation, DistributorNodeInstrumentation, StorageNodeInstrumentation } from './instrumentations'
+
+// For troubleshooting, set the log level to DiagLogLevel.DEBUG
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.INFO)
+
+async function checkApmServer(apmServerUrl: string, retryInterval = 6000, retryAttempts = 10): Promise<void> {
+  let attempts = 0
+
+  const wait = (ms: number) => {
+    return new Promise((resolve) => setTimeout(resolve, ms))
+  }
+
+  while (attempts < retryAttempts) {
+    attempts++
+    try {
+      const response = await axios.get(apmServerUrl)
+
+      if (response.status === 200) {
+        console.log('APM server is running.')
+        return
+      } else {
+        console.log('APM server may not be running correctly.')
+      }
+    } catch (error: any) {
+      console.error(`Failed to reach APM server, attempt ${attempts}:`, error.message)
+      if (attempts < retryAttempts) {
+        await wait(retryInterval)
+      } else {
+        console.log(`Failed to reach APM server after ${attempts} attempts.`)
+        process.exit(1)
```

Comment:
I think we should drop waiting for APM to be reachable entirely, to avoid potentially restarting loop if it ever restart or goes down. Worst case we will see logs indicating otl export failure and queue will get filled.

Reply:
Addressed